stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt

```diff
@@ -99,21 +99,21 @@ public class ChannelClient internal constructor(
         return client.events().filter(this::isRelevantForChannel)
     }
 
-    override fun subscribe(listener: (event: ChatEvent) -> Unit): Disposable {
+    override fun subscribe(listener: ChatClient.ChatEventListener<ChatEvent>): Disposable {
         return client.subscribe(filterRelevantEvents(listener))
     }
 
     override fun subscribeFor(
         vararg eventTypes: String,
-        listener: (event: ChatEvent) -> Unit,
+        listener: ChatClient.ChatEventListener<ChatEvent>,
     ): Disposable {
         return client.subscribeFor(*eventTypes, listener = filterRelevantEvents(listener))
     }
 
     override fun subscribeFor(
         lifecycleOwner: LifecycleOwner,
         vararg eventTypes: String,
-        listener: (event: ChatEvent) -> Unit,
+        listener: ChatClient.ChatEventListener<ChatEvent>,
```

Comment:
It's weird that in `ChannelClient` we use an interface nested in `ChatClient` - it should just be a top level interface instead.